core/field_angle.js

```diff
@@ -344,7 +335,14 @@ Blockly.FieldAngle.prototype.classValidator = function(text) {
   if (n > Blockly.FieldAngle.WRAP) {
     n -= 360;
   }
-  return String(n);
+  return n;
+};
+
+Blockly.FieldAngle.prototype.doValueUpdate_ = function(newValue) {
+  Blockly.FieldAngle.superClass_.doValueUpdate_.call(this, newValue);
+  // Always render the input angle.
+  this.text_ = String(newValue);
+  this.forceRerender();
```

Comment:
You call `forceRerender` in `Field.setValue`, just after `doValueUpdate_`.  Why the duplication?

Reply:
Not sure why this ended up down here, it should be inside the onMouseMove function. 

Fixed.